tokio-trace-env-logger/examples/hyper-echo.rs

```diff
@@ -106,7 +106,7 @@ fn main() {
     server_span.clone().enter(|| {
         let span2 = server_span.clone();
         let server = Server::bind(&addr)
-            .serve(move || span2.enter(|| service_fn(echo)))
+            .serve(move || span2.clone().enter(|| service_fn(echo)))
```

Comment:
I'm guessing the clone operation will need to be a regular thing for supporting futures 0.1 (until await/async lands)?

Reply:
(Yeah, on more thought, that seems to be the case, because with async/await you can rely on scope, but you can't with closure-based transformations.)

Reply:
the clone here has to be added because the `Span` type is consumed to enter a span now, that changed in this branch.

Reply:
Yep—I think I understood the implications of this design in theory, but it didn't click until I saw it in practice.